packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -132,12 +139,14 @@ const App = (props) => {
     // Handle creating a new basket if there isn't one already assigned to the current
     // customer.
     const {data: customer} = useCurrentCustomer()
+
     const {data: baskets} = useCustomerBaskets(
         {parameters: {customerId: customer.customerId}},
-        {enabled: !!customer.customerId && !isServer}
+        {enabled: !!customer.customerId && !isServer, keepPreviousData: true}
```

Comment:
We need to keep previous basket data so we know if a guest user has item in their basket. Without this config, the basket will turn undefined when the cache is invalidated

Reply:
This is not likely to affect anywhere else since this is the only place that use hook directly, the rest of the app use `useCurrentBasket` to access the basket

Reply:
😅 kinda intense that we're doing cache invalidation logic tricks. In an ideal world (not saying we should do this) we would store the state in redux or similar and then wait to update until the callback where the mutation succeeds. Can we do something similar?

Reply:
Perhaps a better way would be to add a `useState` that intentionally holds this data and separates it from the cached data so that it is only updated intentionally and doesn't happen implicitly / magically via cache internals

Reply:
this has nothing to do with cache invalidation, this is 

> "We don't care about loading state! Only update data when we got network response."

The approach seems okay to me, comparing adding complexity for skipping loading state.

Reply:
agreed with Kevin's point, the config is to tell react query to keep the previous data. We need the previous data to be able to carry out the condition. It does not affect loading states so to speak